src/query/xasl_cache.c

```diff
@@ -2212,7 +2316,11 @@ xcache_check_recompilation_threshold (THREAD_ENTRY * thread_p, XASL_CACHE_ENTRY
 	    }
 	  else
 	    {
-	      recompile = true;
+	      /* mark the entry as requst recompile, the client will request a prepare */
```

Comment:
When there are multiple queries on same class that needs recompiling, catalog_update_class_info is called for each class. I think cls_info_p->cit_tot_pages can be checked and catalog_update_class_info can be avoided if it looks similar.